src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -205,11 +205,22 @@ public PackageBuilder GetPackageBuilder(IPackTaskRequest<IMSBuildItem> request)
                     .ToDictionary(msbuildItem => msbuildItem.Identity,
                     msbuildItem => msbuildItem.GetProperty("ProjectVersion"), PathUtility.GetStringComparerBasedOnOS());
             }
+            var nuGetFrameworkComparer = new NuGetFrameworkFullComparer();
+            var frameworksWithSuppressedDependencies = new HashSet<NuGetFramework>(nuGetFrameworkComparer);
+            if (request.FrameworksWithSuppressedDependencies != null && request.FrameworksWithSuppressedDependencies.Any())
```

Comment:
FrameworksWithSuppressedDependencies is an array so you can use `Length` field instead of `Any()` which works on `Enumerable` interface even for a list or array